### PR TITLE
Fix flaky `relative weekdate or date: date before this week` test

### PR DESCRIPTION
**Pull request by josefarias**
This would fail a week from new year's because `CalendarDate.occursOnSameYearAs(date)` compares the literal year values.

We could also account for this edge case in the code instead. But I don't think it's worth complicating.

### CODE REVIEW

File: test/javascripts/src/relative_date_test.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import LocalTime from "local_time"
 
-const { addTimeEl, assert, defer, getText, setText, test, testAsync, testGroup, triggerEvent } = LocalTime.TestHelpers
+const { addTimeEl, assert, defer, getText, testAsync, testGroup, mockNow } = LocalTime.TestHelpers
```
Comment (josefarias) [inline test/javascripts/src/relative_date_test.js:3]: We're not using these
